### PR TITLE
Restrict external data access

### PR DESCRIPTION
**Pull request by crspeller**
Restricts external data access to only when the AI bot is responding to a DM.
External data access is:
- Fetching posts from out of channel.
- Fetching data from GitHub
- Fetching user profile data.

The only exception is user profile data is fetched when SAML or LDAP is enabled for the fetched user and all fields are controlled by LDAP or SAML.

### CODE REVIEW

File: server/ai/conversation.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ type Post struct {
 }
 
 type ConversationContext struct {
+	BotID            string
```
Comment (jupenur) [inline server/ai/conversation.go:28]: You never assign this, meaning `IsDMWithBot` returns true for *all* DM channels. Please add unit tests and return false from `IsDMWithBot` when the bot ID is empty.

File: server/built_in_tools.go [hunk 2]
```diff
@@ -38,6 +38,13 @@ func (p *Plugin) toolResolveLookupMattermostUser(context ai.ConversationContext,
 		return "", errors.Wrap(err, "failed to lookup user")
 	}
 
+	// If this isn't a DM fail for users who are not SSO users.
+	if !context.IsDMWithBot() {
+		if user.AuthService != model.UserAuthServiceSaml && user.AuthService != model.UserAuthServiceLdap {
```
Comment (jupenur) [inline server/built_in_tools.go]: At least with LDAP it looks like switching from email auth does not immediately update the profile, meaning you could have `user.AuthService == model.UserAuthServiceLdap` and user attributes that were manually set while on email auth. I *think* that's the same for SAML too, but I need to confirm.

Anyway, this check isn't good enough.

**Review by jupenur — CHANGES_REQUESTED** [review]

File: server/ai/conversation.go (hunk 1 above)
Comment (crspeller) [reply server/ai/conversation.go:28]: Good catch. Updated to prevent this.

File: server/built_in_tools.go (hunk 2 above)
Comment (crspeller) [reply server/built_in_tools.go]: Don't see a way to check for this case. Removing the exception for now.

**Review by jupenur — APPROVED** [review]
Other than the issue concerning DMs I shared in-thread, LGTM.
